docs/usage.rst

```diff
@@ -452,6 +452,30 @@ functions that leverage this capability; builtins including, but not limited to:
 (for examples, see :ref:`sec_max_min`) and :func:`sorted`.
 
 
+Version Equality
+----------------
+
+Note that when comparing two versions, the ``build`` part is ignored::
+
+    >>> v = semver.VersionInfo.parse("1.2.3-rc4+1e4664d")
+    >>> v == "1.2.3-rc4+dedbeef"
+    True
+
+This also applies when a :class:`semver.VersionInfo` is a member of a set, or a
+dictionary key::
+
+    >>> d = {}
+    >>> v1 = semver.VersionInfo.parse("1.2.3-rc4+1e4664d")
+    >>> v2 = semver.VersionInfo.parse("1.2.3-rc4+dedbeef")
+    >>> d[v1] = 1
```

Comment:
Sorry, I'm not sure what are you trying to show here. Probably I've overlooked something. Could you enlighten me? :-)

Reply:
I was trying to illustrate that semver-style equality means that if two versions are equal with different `build`s, then python won't be able to distinguish them as distinct dict keys or set members.

This may be helpful or harmful to authors depending what they're trying to do, but in any case it's surprising in Python that `Obj(args) == Obj(different_args)` (except for primitives). There isn't too much precedent for it. So I was trying to think of important implications of this design and illustrate those to reduce the surprise.

Reply:
Ahh, now I got it. Thanks! :+1: